application/src/test/java/org/thingsboard/server/service/state/DefaultDeviceStateServiceTest.java

```diff
@@ -125,4 +140,81 @@ private void process(Map<EntityKeyType, Map<String, TsValue>> latest, long defau
+        ReflectionTestUtils.setField(service, "defaultActivityStatsIntervalInSec", 60);
+        ReflectionTestUtils.setField(service, "defaultInactivityTimeoutMs", 1);
+        ReflectionTestUtils.setField(service, "initFetchPackSize", 10);
+
+        Mockito.when(entityQueryRepository.findEntityDataByQueryInternal(Mockito.any())).thenReturn(new PageData<>());
+
+        service.init();
+        var tenantId = new TenantId(UUID.randomUUID());
+        var tpi = TopicPartitionInfo.builder().myPartition(true).build();
+        Mockito.when(partitionService.resolve(ServiceType.TB_CORE, tenantId, deviceId)).thenReturn(tpi);
+
+        var deviceIdInfo = new DeviceIdInfo(tenantId.getId(), null, deviceId.getId());
+
+        Mockito.when(deviceService.findDeviceIdInfos(Mockito.any()))
+                .thenReturn(new PageData<>(List.of(deviceIdInfo), 0, 1, false));
+
+        Method method = AbstractPartitionBasedService.class.getDeclaredMethod("initStateFromDB", Set.class);
+        method.setAccessible(true);
+        method.invoke(service, Collections.singleton(tpi));
+
+        service.onAddedPartitions(Collections.singleton(tpi));
+
+        DeviceState deviceState = DeviceState.builder().build();
+
+        DeviceStateData deviceStateData = DeviceStateData.builder()
+                .tenantId(tenantId)
+                .deviceId(deviceId)
+                .state(deviceState)
+                .metaData(new TbMsgMetaData())
+                .build();
+
+        service.deviceStates.put(deviceId, deviceStateData);
+
+        service.getPartitionedEntities(tpi).add(deviceId);
+
+        service.onDeviceActivity(tenantId, deviceId, System.currentTimeMillis());
+
+        Mockito.verify(telemetrySubscriptionService, Mockito.times(1)).saveAttrAndNotify(Mockito.any(), Mockito.eq(deviceId), Mockito.any(), Mockito.eq("active"), Mockito.eq(true), Mockito.any());
+
+        Thread.sleep(1);
+
+        service.checkStates();
+
+        Mockito.verify(telemetrySubscriptionService, Mockito.times(1)).saveAttrAndNotify(Mockito.any(), Mockito.eq(deviceId), Mockito.any(), Mockito.eq("active"), Mockito.eq(false), Mockito.any());
+
+        Mockito.reset(telemetrySubscriptionService);
+
+        service.onDeviceInactivityTimeoutUpdate(tenantId, deviceId, System.currentTimeMillis() - deviceState.getLastActivityTime() + 1000);
+
+        Mockito.verify(telemetrySubscriptionService, Mockito.times(1)).saveAttrAndNotify(Mockito.any(), Mockito.eq(deviceId), Mockito.any(), Mockito.eq("active"), Mockito.eq(true), Mockito.any());
+
+        Thread.sleep(2000);
+
+        service.checkStates();
+
+        Mockito.verify(telemetrySubscriptionService, Mockito.times(1)).saveAttrAndNotify(Mockito.any(), Mockito.eq(deviceId), Mockito.any(), Mockito.eq("active"), Mockito.eq(false), Mockito.any());
+
+        Mockito.reset(telemetrySubscriptionService);
+
+        service.onDeviceInactivityTimeoutUpdate(tenantId, deviceId, 2000);
```

Comment:
we already waited 2000ms before (Thread.sleep). Setting inactivite timeout 2000 have to report Active=false, but test if answering true